src/industry_cmd.cpp

```diff
@@ -1126,9 +1126,15 @@ static void ProduceIndustryGoods(Industry *i)
 		uint32 r;
 		uint num;
 		if (Chance16R(1, 14, r) && (num = indsp->number_of_sounds) != 0 && _settings_client.sound.ambient) {
```

Comment:
not strictly your issue, but i'd want to see num (and it's declaration) moved inside the if statement

Reply:
Should I do that as part of my PR?

Reply:
Also, why should it be moved inside? Is it because it is bad practice to declare in the statement?

Reply:
Bad practice to declare variables in a scope above where they're used. Also, equality in if statements is unpleasant in all but the simplest of cases. Not sure why an extra variable is needed at all, tbh.

`indsp->random_sounds[((r >> 16) * indsp->number_of_sounds) >> 16]` would work just fine, though I can't say I understand what it's actually doing with all the bitshifts. Probably some remnant of the original TTD ASM.

Reply:
Is `indsp->number_of_sounds` a constant?

Reply:
regardless, it's not modified in this function

Reply:
Push a commit removing it. Could you please verify it is what you wanted?